lib/internal/bootstrap/node.js

```diff
@@ -305,17 +305,13 @@ if (process.env.NODE_V8_COVERAGE) {
 }
 
 function setupProcessObject() {
+  const { cjsContext } = NativeModule;
   const EventEmitter = NativeModule.require('events');
   const origProcProto = Object.getPrototypeOf(process);
   Object.setPrototypeOf(origProcProto, EventEmitter.prototype);
   EventEmitter.call(process);
-  // Make process globally available to users by putting it on the global proxy
-  Object.defineProperty(global, 'process', {
-    value: process,
-    enumerable: false,
-    writable: true,
-    configurable: true
-  });
+
+  global.process = cjsContext.process = process;
```

Comment:
Why is it necessary here? Before pre-execution, all builtin modules should be able to get this through the parameters passed to `CompileFunctionInContext`

Reply:
Do all environments definitely get pre-execution running? I was just concerned about missing anything out.

Reply:
They do

Reply:
Ok great, I've added this in the latest commit.